lib/improver/threshold.py

```diff
@@ -62,34 +64,41 @@ def __init__(self, threshold, fuzzy_factor=None,
                 False to count points as significant if *above* the threshold.
 
         Raises:
-            ValueError: If a threshold of 0.0 is requested.
+            ValueError: If a threshold of 0.0 is requested when using a fuzzy
+                        factor.
             ValueError: If the fuzzy_factor is not greater than 0 and less
                         than 1.
 
         """
-        if threshold == 0.0:
-            raise ValueError(
-                "Invalid threshold: zero not allowed")
-        self.threshold = threshold
+        # Ensure iterable threshold list provided, even if it's a single value.
+        self.thresholds = thresholds
+        if np.isscalar(thresholds):
+            self.thresholds = [thresholds]
+
         if fuzzy_factor is not None:
             if not 0 < fuzzy_factor < 1:
                 raise ValueError(
                     "Invalid fuzzy_factor: must be >0 and <1: {}".format(
                         fuzzy_factor))
+            if any(threshold == 0 for threshold in self.thresholds):
```

Comment:
Does this work if it is `if 0 in thresholds`?

Reply:
That works, so I'll change it to that.